depthmapX/mainwindow.cpp

```diff
@@ -1815,29 +1818,39 @@ void MainWindow::SelectButtonTriggered()
 {
     m_selected_mapbar_item = ID_MAPBAR_ITEM_SELECT;
     activeQDepthmapView()->OnEditSelect();
+    GLView* glView = getFirstGLView();

```

Comment:
Not sure I like this (or maybe I don't understand?): Can't this code know whether we're in QDepthMapView or GLViewMode and only do the one that's actually used? This whole "go through widgets and see if one is the view I want to work on" pattern seems somewhat insane, and this is going in the same direction, if I am not mistaken?

Reply:
The following (and final in this saga) pull request actually has a main class "MapView" that both QDepthmapView and GLView descend from (And the above piece becomes like [this](https://github.com/orange-vertex/depthmapX/blob/map_opengl_default_window/depthmapX/mainwindow.cpp#L1838) so for MainWindow it doesn't matter which one you're looking at, as long as it's a MapView

Yes, the pattern is insane, and it is jumping through weird hoops, but I wasn't aiming to fix it, as that would require going into the table, scatter and agent views as well...